src/inspector_agent.cc

```diff
@@ -0,0 +1,515 @@
+    "@521e5b7e2b7cc66b4006a8a54cb9c4e57494a5ef/inspector.html?"
+    "experiments=true&v8only=true&ws=localhost:%d/node\n", port, port);
+}
+
+bool AcceptsConnection(inspector_socket_t* socket, const char* path) {
+  return strncmp(DEVTOOLS_PATH, path, sizeof(DEVTOOLS_PATH)) == 0;
+}
+
+void DisposeAsyncCb(uv_handle_t* handle) {
+  free(handle);
+}
+
+void DisposeInspector(inspector_socket_t* socket, int status) {
+  free(socket);
+}
+
+void DisconnectAndDisposeIO(inspector_socket_t* socket) {
+  if (socket) {
+    inspector_close(socket, DisposeInspector);
+  }
+}
+
+void OnBufferAlloc(uv_handle_t* handle, size_t len, uv_buf_t* buf) {
+  if (len > 0) {
+    buf->base = reinterpret_cast<char*>(malloc(len));
+    CHECK_NE(buf->base, nullptr);
+  }
+  buf->len = len;
+}
+
+void SendHttpResponse(inspector_socket_t* socket,
+                        const char* response,
+                        size_t len) {
+  const char HEADERS[] = "HTTP/1.0 200 OK\r\n"
+                         "Content-Type: application/json; charset=UTF-8\r\n"
+                         "Cache-Control: no-cache\r\n"
+                         "Content-Length: %ld\r\n"
+                         "\r\n";
+  char header[sizeof(HEADERS) + 20];
+  int header_len = snprintf(header, sizeof(header), HEADERS, len);
+  inspector_write(socket, header, header_len);
+  inspector_write(socket, response, len);
+}
+
+void SendVersionResponse(inspector_socket_t* socket) {
+  const char VERSION_RESPONSE_TEMPLATE[] =
+      "[ {"
+      "  \"Browser\": \"node.js/%s\","
+      "  \"Protocol-Version\": \"1.1\","
+      "  \"User-Agent\": \"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36"
+            "(KHTML, like Gecko) Chrome/45.0.2446.0 Safari/537.36\","
+      "  \"WebKit-Version\": \"537.36 (@198122)\""
+      "} ]";
+  char buffer[sizeof(VERSION_RESPONSE_TEMPLATE) + 128];
+  size_t len = snprintf(buffer, sizeof(buffer),
+                 VERSION_RESPONSE_TEMPLATE, NODE_VERSION);
+  ASSERT(len < sizeof(buffer));
+  SendHttpResponse(socket, buffer, len);
+}
+
+void SendTargentsListResponse(inspector_socket_t* socket) {
+  const char LIST_RESPONSE_TEMPLATE[] =
+      "[ {"
+      "  \"description\": \"node.js instance\","
+      "  \"devtoolsFrontendUrl\": "
+            "\"https://chrome-devtools-frontend.appspot.com/serve_file/"
+            "@4604d24a75168768584760ba56d175507941852f/inspector.html\","
+      "  \"faviconUrl\": \"https://nodejs.org/static/favicon.ico\","
+      "  \"id\": \"%d\","
+      "  \"title\": \"%s\","
+      "  \"type\": \"node\","
+      "  \"webSocketDebuggerUrl\": \"ws://%s\""
+      "} ]";
+  char buffer[sizeof(LIST_RESPONSE_TEMPLATE) + 4096];
+  char title[2048];  // uv_get_process_title trims the title if too long
+  int err = uv_get_process_title(title, sizeof(title));
+  ASSERT_EQ(0, err);
+  size_t len = snprintf(buffer, sizeof(buffer), LIST_RESPONSE_TEMPLATE,
+                        getpid(), title, DEVTOOLS_PATH);
```

Comment:
It's not very likely but `process.title = '... "boom" ...'` would break this, wouldn't it?